### PR TITLE
Fix bz2 unzip function errors

### PR DESCRIPTION
**Pull request by simonrp84**
The bz2 unzip function in `readers/utils.py` doesn't close the BZ2 file handler, which on some platforms leads to permission errors when `os.remove` tries to delete the unclosed files.
This PR wraps the BZ2 handler in a `with closing()` statement to prevent this error.

### CODE REVIEW

**Comment by mraspaud** [comment]
Maybe related to this https://github.com/pytroll/satpy/issues/2183 ?

File: satpy/readers/utils.py [hunk 1]
```diff
@@ -247,17 +247,17 @@ def unzip_file(filename, prefix=None):
...
-            try:
-                ofpt.write(bz2file.read())
-            except IOError:
-                import traceback
-                traceback.print_exc()
-                LOGGER.info("Failed to read bzipped file %s", str(filename))
-                os.remove(tmpfilepath)
-                return None
-        return tmpfilepath
+        with closing(bz2.BZ2File(filename)) as bz2file:
```
Comment (mraspaud) [inline satpy/readers/utils.py:250]: would the simpler `with bz2.open(filename) as bz2file:` also work?

Comment (simonrp84) [reply satpy/readers/utils.py]: Done!

**Comment by mraspaud** [comment]
Looks good, do you think the block above could also drop the `closing` or would the raising in the except block be a problem?

**Comment by simonrp84** [comment]
@mraspaud I *think* it should be OK to remove that `closing` but that's not my code originally so I'm not 100% sure what the original purpose was. I'll remove it as the tests pass OK for this.

**Review by mraspaud — COMMENTED** [review]
Is there a test that checks that the file can be removed now?

Overall, I think using the tmp_path fixture would make it a bit cleaner, but that's optional

**Comment by mraspaud** [comment]
I think there has been quite some work merged for this already, so I'm closing this PR for now. Feel free to reopen if it turns out that problem isn't solved.

**Pull request closed without merging**